apps/omg_watcher/lib/omg_watcher/exit_processor.ex

```diff
@@ -268,64 +269,46 @@ defmodule OMG.Watcher.ExitProcessor do
 
   def handle_call({:finalize_in_flight_exits, finalizations}, _from, state) do
     _ = if not Enum.empty?(finalizations), do: Logger.info("Recognized ife finalizations: #{inspect(finalizations)}")
-
-    case Core.finalize_in_flight_exits(state, finalizations) do
-      {:ok, state, db_updates} ->
-        {:reply, {:ok, db_updates}, state}
-
-      {:unknown_piggybacks, unknown_piggybacks} ->
-        _ = Logger.error("Outputs not piggybacked: #{inspect(unknown_piggybacks)}")
-        {:stop, :unknown_piggybacks, state}
-
-      {:unknown_in_flight_exit, unknown_ifes} ->
-        _ = Logger.error("Unknown in-flight exits: #{inspect(unknown_ifes)}")
-        {:stop, :unknown_in_flight_exit, Elixir.Agent.Server, state}
-    end
+    {:ok, state, db_updates} = Core.finalize_in_flight_exits(state, finalizations)
+    {:reply, {:ok, db_updates}, state}
   end
 
   @doc """
   Combine data from `ExitProcessor` and `OMG.State` to figure out what to do about exits
   """
   def handle_call(:check_validity, _from, state) do
-    {state1, request} = prepare_validity_check(state)
-    {chain_status, events} = Core.invalid_exits(request, state1)
-    {:reply, {chain_status, events}, state}
+    state1 =
+      %ExitProcessor.Request{}
```

Comment:
is it a part of some greater refactor? If no then why not pass default argument?

Reply:
ah right, this is some brain fart here. 

But I kept the explicit `%ExitProcessor.Request` to be piped to `fill_request_with_data`, because otherwise it is much less readable - I want to emphasize that we're starting off with a clean request struct and then piping it through the process.

But for `update_with_ife_txs_from_blocks` the argument makes no sense